### DISCUSSION TITLE
经常抛出异常日志Failed to decode video, please play PAG after user gesture. Or your can load a software decoder to decode the video.

### DISCUSSION THREAD

**Discussion by wang-jiyu** [Tencent/libpag#1749] [Q&A] [answered]
## 【版本信息】

4.1.8.1

## 【平台信息】

Web 浏览器

## 【预期的表现】
不想有异常抛出，因为我们会

## 【Demo及附件】

```[tasklist]
### Tasks
```

**Comment by wang-jiyu** [comment]
预期表现没有写完：关于类似的情况如果咱们没有办法做到兼容可以使用告警类提示抛出，别抛出异常。现在我们的后面监控每天几万条都是这类错误

**Comment by domchen** [comment, marked as answer]
遇到这个情况是严重错误，画面会直接不播放，不解决根本没法正常使用，是必须抛出异常的。你如果是运行在自己可以控制的webview，开启一下允许视频自动播放的属性。如果是外部的参考这个文档里的例子注入软件解码器ffavc：https://pag.art/docs/web-features.html 再或者从素材入手，不要导出带BMP预合成的PAG文件也不会有这个问题。
